v2/tools/generator/internal/codegen/pipeline/transform_validated_floats.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+)
+
+// TransformValidatedFloatsStageID is the unique identifier of this stage
+const (
+	TransformValidatedFloatsStageID = "transformValidatedFloats"
+)
+
+func TransformValidatedFloats() *Stage {
+	stage := NewStage(
+		TransformValidatedFloatsStageID,
+		"Transform validated 'spec' float type values to validated integer types for compatibility with controller-gen",
+		func(ctx context.Context, state *State) (*State, error) {
+
+			definitions := state.Definitions()
+
+			result, err := getFloatTransformations(definitions)
+			if err != nil {
+				return nil, err
+			}
+
+			remaining := definitions.Except(result)
+			result.AddTypes(remaining)
+
+			return state.WithDefinitions(result), nil
+		})
+
+	stage.RequiresPrerequisiteStages(RemoveStatusPropertyValidationsStageID)
+
+	return stage
+}
+
+func getFloatTransformations(definitions astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, error) {
+	visitor := astmodel.TypeVisitorBuilder{
+		VisitValidatedType: visitValidatedType,
+		VisitPrimitive:     transformFloatToInt,
+	}.Build()
+
+	return visitor.VisitDefinitions(astmodel.FindSpecConnectedDefinitions(definitions))
+}
+
+func visitValidatedType(this *astmodel.TypeVisitor, validated *astmodel.ValidatedType, ctx interface{}) (astmodel.Type, error) {
+	return astmodel.IdentityVisitOfValidatedType(this, validated, true) // Pass ctx so that transformFloatToInt can use it
+}
+
+// transformFloatToInt transforms all the validated FloatTypes to IntegerTypes
+func transformFloatToInt(this *astmodel.TypeVisitor, prim *astmodel.PrimitiveType, ctx interface{}) (astmodel.Type, error) {
+	validated, _ := ctx.(bool)
```

Comment:
Nitpick - need to check to see whether `ctx` really was a **bool**, more to avoid issues with future maintenance than anything else. 

Can easily be done by including a local `ok` here and then checking that as well in the condition just below.